test/exe/BUILD

```diff
@@ -69,6 +69,7 @@ envoy_cc_test(
         "//source/common/api:api_lib",
         "//source/exe:envoy_main_common_with_core_extensions_lib",
         "//source/exe:platform_impl_lib",
+        "//source/extensions/clusters/logical_dns:logical_dns_cluster_lib",
```

Comment:
Hmm... I wonder if this target can be built w/o reliance on source/extensions.

Reply:
no, there's already a number of registered extensions such as source/extensions/transport_sockets/raw_buffer/config.h that envoy won't build without

Reply:
that said this is why we have the visibility rules and an extra presubmit around extensions not generally being visible to core tests.  so it's a pretty well cross-checked system :-)